splunk-otel-android/src/main/java/com/splunk/rum/FileSender.java

```diff
@@ -39,11 +41,11 @@ class FileSender {
     private final BandwidthTracker bandwidthTracker;
     private final RetryTracker retryTracker;
 
-    private FileSender(Builder builder) {
-        this.sender = builder.sender;
+    private FileSender(Builder builder, RetryTracker retryTracker) {
```

Comment:
This feels weird to me. Maybe rather than the builder passing it in, maybe the `FileSender` constructor should create it based on the fields on the builder? Maybe that violates DI...but the mixed-mode constructor here (builder param with bonus param) feels odd.